code/modules/surgery/organs/stomach.dm

```diff
@@ -58,6 +62,19 @@
 /obj/item/organ/stomach/get_availability(datum/species/S)
 	return !(NOSTOMACH in S.inherent_traits)
 
+/**
+ * Triggers on bitting the food item before it is consumed.
+ *
+ * Checks if the stomach is useable, if so adds the food reagents.
+ */
+/obj/item/organ/stomach/proc/on_bite(datum/source, mob/living/target, mob/living/user, bitecount, bitesize)
```

Comment:
Besides this not fixing the issue because the stomach will metabolize the reagents even when outside of a body, this is also a ridiculous way of doing this; you bite into the stomach so reagents are suddenly added to it? No way. 

The entire useable = thing is also unnecesary as this is already handled by edible component code.

Reply:
useable is a organ level var and is already set to false on after bite, so i am using functionality that already exists, and is for this intended reason.
what is the flag for first bite in edible code? I am not seeing one, no call backs on that, no signals on that.

Reply:
`/obj/item/organ/proc/OnEatFrom(eater, feeder)
	useable = FALSE //You can't use it anymore after eating it you spaztic`

Reply:
`OnEatFrom` is the `after_eat` callback that happens after the the qdel of the item. this would not be a valid usage of this logic.

Reply:
no its not, thats on_consume.

Reply:
`on_consume` happens in `TakeBite` after all checks
`after_eat` is at the end of `TakeBite` after all checks and after `on_consume`.

Reply:
bruh read the code